community/kernel/src/main/java/org/neo4j/kernel/impl/api/OperationsFacade.java

```diff
@@ -366,14 +365,15 @@ public Object nodeGetProperty( long nodeId, int propertyKeyId ) throws EntityNot
     }
 
     @Override
-    public RelationshipIterator nodeGetRelationships( long nodeId, Direction direction, int... relTypes )
+    public RelationshipIterator nodeGetRelationships( long nodeId, Direction direction, int[] relTypes )
             throws EntityNotFoundException
     {
         statement.assertOpen();
         try ( Cursor<NodeItem> node = dataRead().nodeCursorById( statement, nodeId ) )
         {
-            return new CursorRelationshipIterator(
-                    node.get().relationships( direction( direction ), deduplicate( relTypes ) ) );
+            return new CursorRelationshipIterator( dataRead()
+                    .nodeGetRelationships( statement, node.get(), direction( direction ),
+                            asSet( relTypes, t -> t >= 0 ) ) );
```

Comment:
Oh we did? Alright, then keeping it is alright, agreed.

Keep in mind that normally the number of types is very few and so scanning the array will most likely be faster than asking the set too.

No we didn't/don't, I meant that we'd quite like to do that in the near future.